### ISSUE TITLE
 compile with the mode aot 

### ISSUE THREAD

**Issue by kgiroux** [angular/material2#1923]
#### Bug, feature request, or proposal:
Try to compile with the mode aot

 Cannot determine the module for class FocusTrap in node_modules/@angular/material/core/a11y/focus-trap.d.ts!
at analyzeNgModules (/node_modules/@angular/compiler/bundles/compiler.umd.js:13079:17)
    at OfflineCompiler.compileModules (/node_modules/@angular/compiler/bundles/compiler.umd.js:13115:20)
    at CodeGenerator.codegen (/node_modules/@angular/compiler-cli/src/codegen.js:58:30)
    at codegen (/node_modules/@angular/compiler-cli/src/main.js:7:81)
    at Object.main (/@angular/tsc-wrapped/src/main.js:30:16)
    at Object.<anonymous> (/@angular/compiler-cli/src/main.js:16:9)
    at Module._compile (module.js:573:32)
    at Object.Module._extensions..js (module.js:582:10)
    at Module.load (module.js:490:32)
    at tryModuleLoad (module.js:449:12)

#### What is the expected behavior?
Compile with aot option

**Comment by jelbourn** [comment]
I think you're having this issue because you used `yarn` instead of `npm install`. For some reason, `yarn` doesn't bring in all of the files in the package correctly.

**Comment by kgiroux** [comment]
I didn't use yarn...

**Comment by jelbourn** [comment]
@kgiroux Make sure you have a decently up-to-date `node` and `npm`, wipe out `node_modules` and re-install. 

Here is a sample app that uses alpha.10 with AoT without any issues:
https://github.com/jelbourn/material2-app

**Comment by Magnetars** [comment]
I had the same issue, it got fixed (for AoT compilation, haven't tested running it yet) when I forced installing version "@angular/material": "2.0.0-alpha.10".  

try `npm install @angular/material@2.0.0-alpha.10 --save`

Seems an older version gets installed if you don't specify the version manually. That's an npm view of the versions.

http://prntscr.com/d9xzdu
